libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk.pass.cpp

```diff
@@ -42,25 +42,30 @@ __device__ void test()
 
   // TEST: Add i to buffer[i]
   alignas(16) __shared__ int smem_buffer[buf_len];
-  __shared__ barrier* bar;
+#if _CCCL_CUDA_COMPILER(CLANG)
+  __shared__ char barrier_data[sizeof(barrier)];
+  barrier& bar = cuda::std::bit_cast<barrier>(barrier_data);
+#else // ^^^ _CCCL_CUDA_COMPILER(CLANG) ^^^ / vvv !_CCCL_CUDA_COMPILER(CLANG)
+  __shared__ barrier bar;
+#endif // !_CCCL_CUDA_COMPILER(CLANG)
```

Comment:
Q: So this was just a plain bug before, since `bar` did not point anywhere?

Also: why is the bitcast needed for clang? The non-clang code path is what I think users should be able to write. What is worked around here?

Reply:
urgh seems like it did not properly save